eslint.config.js

```diff
@@ -0,0 +1,99 @@
+import hypothesis from 'eslint-config-hypothesis';
+import jsxA11y from 'eslint-plugin-jsx-a11y';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+  {
+    ignores: [
+      '.tox/**/*',
+      '.yalc/**/*',
+      '.yarn/**/*',
+      'build/**/*',
+      '**/vendor/**/*.js',
+      '**/coverage/**/*',
+      'docs/_build/*',
+      'dev-server/static/**/*.js',
+    ],
+  },
+  ...hypothesis,
+  ...tseslint.configs.recommended,
+  {
+    plugins: {
+      'jsx-a11y': jsxA11y,
+    },
+    rules: {
+      'prefer-arrow-callback': [
+        'error',
+        {
+          allowNamedFunctions: true,
+        },
+      ],
+
+      'object-shorthand': ['error', 'properties'],
+      'react/prop-types': 'off',
+      '@typescript-eslint/no-unused-vars': 'error',
+      'no-use-before-define': 'off',
+
+      '@typescript-eslint/no-use-before-define': [
+        'error',
+        {
+          functions: false,
+          typedefs: false,
+          ignoreTypeReferences: false,
+        },
+      ],
+
+      '@typescript-eslint/ban-ts-comment': 'off',
+      '@typescript-eslint/no-empty-function': 'off',
+      '@typescript-eslint/no-explicit-any': 'off',
+      '@typescript-eslint/no-non-null-assertion': 'off',
+      '@typescript-eslint/no-this-alias': 'off',
+      '@typescript-eslint/consistent-type-assertions': 'error',
+      '@typescript-eslint/consistent-type-imports': 'error',
+
+      // Mocha
+      'mocha/no-mocha-arrows': 'off',
+      'mocha/no-setup-in-describe': 'off',
+      'mocha/max-top-level-suites': 'off',
+      'mocha/consistent-spacing-between-blocks': 'off',
+      'mocha/no-top-level-hooks': 'off',
+      'mocha/no-sibling-hooks': 'off',
+      'mocha/no-identical-title': 'off',
```

Comment:
Enabling mocha plugin using the old ESLint config format didn't enable these rules apparently, while they get enabled now [here](https://github.com/hypothesis/frontend-toolkit/pull/50/files#diff-51bf7f3f79705acd83bbab98a43b506bbce1e535b73bb11f0ab5e79c5c3ad403R9).

I couldn't find a way to enable mocha's plugin without these rules being enabled, so I simply disabled them. We can revisit later.

Reply:
Are we going to need these same overrides in other projects? If so we could push these upstream into eslint-config-hypothesis.

Reply:
I moved them to `eslint-config-hypothesis`. See https://github.com/hypothesis/frontend-toolkit/pull/50